spikeextractors/MultiRecordingChannelExtractor.py

```diff
@@ -0,0 +1,115 @@
+from .RecordingExtractor import RecordingExtractor
+import numpy as np
+
+# Concatenates the given recordings by channel
+
+class MultiRecordingChannelExtractor(RecordingExtractor):
+    def __init__(self, recordings, groups=None):
+        RecordingExtractor.__init__(self)
+        self._RXs = recordings
+        self._all_channel_ids = []
+        self._channel_map = {}
+
+        # Sampling frequency based off the initial extractor
+        self._first_recording = recordings[0]
+        self._sampling_frequency = self._first_recording.get_sampling_frequency()
+        self._num_frames =  self._first_recording.get_num_frames()
+
+        # Test if all recording extractors have same sampling frequency
+        for i, recording in enumerate(recordings[1:]):
+            sampling_frequency = recording.get_sampling_frequency()
+            if (self._sampling_frequency != sampling_frequency):
+                raise ValueError("Inconsistent sampling frequency between extractor 0 and extractor " + str(i + 1))
+
+        #set channel map for new channel ids to old channel ids
+        new_channel_id  = 0
+        for r_i, RX in enumerate(self._RXs):
+            channel_ids = RX.get_channel_ids()
+            for channel_id in channel_ids:
+                self._all_channel_ids.append(new_channel_id)
+                self._channel_map[new_channel_id] = {'rx': r_i, 'channel': channel_id}
+                new_channel_id += 1
```

Comment:
Maybe here you could check if the channels have duplicates. For instance, if you concatenated 2 recordings with channels: [0,1,2,3], [8,9,10,11] the channel ids should be maintained

Reply:
I would rather remap I think just to keep the behaviour consistent